sepal_ui/mapping/mapping.py

```diff
@@ -785,21 +689,125 @@ def get_viz_params(image):
 
         return props
 
+    def remove_layer(self, key):
+        """
+        Remove a layer based on a key. The key can be, a Layer object, the name of a layer or the index in the layer list
+
+        Args:
+            key (Layer, int, str): the key to find the layer to delete
+        """
+
+        if isinstance(key, (int, str, ipl.Layer)):
```

Comment:
from what I understand about `isinstance`, it's already doing that by design: 
```python 
class A: 
    pass

class B(A):
   pass

class C(B):
    pass

print(isinstance(C(), C)) # True
print(isinstance(C(), B)) # True
print(isinstance(C(), A)) # True
```

I checked in ipyleaflet and all layer classes are inherithing from `Layer`

Reply:
This is what I get from my side

```Python
from ipyleaflet import Layer, TileLayer

print(isinstance(TileLayer, Layer)) # False
print(issubclass(TileLayer, Layer)) # True

```

Reply:
Ok I get it, `isinstance` is used by instances, in my example `TileLayer` is a class not the instance, so that's why is `False` in my example. So as the key would be a instance, it will work.

Reply:
that's because you're testing the class and not an object:
```python
from ipyleaflet import basemaps, basemap_to_tiles, Layer

dark_matter_layer = basemap_to_tiles(basemaps.CartoDB.DarkMatter)
print(type(dark_matter_layer)) #ipyleaflet.leaflet.TileLayer
print(isinstance(dark_matter_layer, Layer)) # True
```

This will be False as well: 
```python
print(isinstance(TileLayer, TileLayer)) # False
```